### DISCUSSION TITLE
Despite the docs (and what I can glance from the code), jpeg appears to just work for background images?

### DISCUSSION THREAD

**Discussion by cnschn** [kovidgoyal/kitty#7636] [Q&A] [answered]
I was messing around with some scripts to randomly set the background image for my kitty OS windows, and completely forgot to check for the file type. To my surprise, everything worked just fine when passing a jpeg to the `set-background-image` remote command. Does anyone have an idea why that would be the case? I'm not familiar with the code base at all, but I guess [this line here](
https://github.com/kovidgoyal/kitty/blob/master/kitty/rc/set_background_image.py#L94) should absolutely complain?

I'm on arch, using the `extra/kitty` package.

```
❯ kitty --version
kitty 0.35.2 created by Kovid Goyal

❯ identify test.jpeg 
test.jpeg JPEG 1920x1080 1920x1080+0+0 8-bit sRGB 90145B 0.000u 0:00.000

❯ kitty @ set-background-image test.jpeg 

# no errors, kitty is now very pretty
```

**Comment by kovidgoyal** [comment, marked as answer]
https://github.com/kovidgoyal/kitty/commit/b17c2dd06bca03e2f367c9cc08524888e647b45c

**Reply by cnschn** [reply]
Awesome, thanks for the clarification. I'm still confused about the `message_to_kitty()` method for this command - I assume that's not really the implementation for the command? Because it still contains the check for PNG files only and should fail for anything else.

**Comment by kovidgoyal** [comment]
Nowadays, the implementation is in Go.

**Comment by cnschn** [comment]
If it's relevant for anyone else - this was changed in version `0.27.0`, so from that version on non-PNG files work for `set-background-image`.
